FF125 Element.currentCSSZoom add docs (#33427)
mdn/content: files/en-us/web/api/element/currentcsszoom/index.md

---
title: "Element: currentCSSZoom property"
short-title: currentCSSZoom
slug: Web/API/Element/currentCSSZoom
page-type: web-api-instance-property
browser-compat: api.Element.currentCSSZoom
---

{{APIRef("DOM")}}

> The **`currentCSSZoom`** read-only property of the {{domxref("Element")}} interface is the "effective" [CSS `zoom`](/en-US/docs/Web/CSS/zoom) of an element, taking into account the zoom applied to the element and all its parent elements.

Author: ```suggestion
The **`currentCSSZoom`** read-only property of the {{domxref("Element")}} interface provides the "effective" [CSS `zoom`](/en-US/docs/Web/CSS/zoom) of an element, taking into account the zoom applied to the element and all its parent elements.
```

> The value calculated by multiplying the `zoom` value of the element by the zoom values to all its parents.

Author: ```suggestion
The value calculated by multiplying the `zoom` value of the element by the zoom values of all its parents.
```

For example, if three elements with zoom values of 2, 1.5, and 3, are nested within each other, the most deeply nested element will have a `currentCSSZoom` value of 9.

> If the element is not rendered, for example because `display: none` is set on it, then the `currentCSSZoom` is set to 1.

Author: Clarification of what "rendered" means in this case:
```suggestion
If the element doesn't have a CSS box, for example because `display: none` is set on the element or one of its parents, then the `currentCSSZoom` is set to 1.
```

> The value calculated by multiplying the `zoom` value of the element by the zoom values to all its parents.
> For example, if three elements with zoom values of 2, 1.5, and 3, are nested within each other, the most deeply nested element will have a `currentCSSZoom` value of 9.
> If the element doesn't have a CSS box, for example because `display: none` is set on the element or one of its parents, then the `currentCSSZoom` is set to 1.
> 
> Note that some APIs, such as {{domxref("Element.getBoundingClientRect()")}}, return values that are relative to the viewport, and hence include the effects of CSS `zoom`.
> Other APIs return values relative to the element itself, which do not include the effects of zooming.
> These include, for example, `client*` properties such as {{domxref("Element.clientHeight")}}, `scroll*()` methods like {{domxref("Element.scroll()")}}, and `offset*` properties such as {{domxref("HTMLElement.offsetHeight")}}.
> The `currentCSSZoom` property can be used to scale these values to adjust for the effects of zooming.

Author: ```suggestion
The value calculated by multiplying the CSS `zoom` values of the element and all of its parents.
For example, if three elements with zoom values of 2, 1.5, and 3, are nested within each other, the most deeply nested element will have a `currentCSSZoom` value of 9.
If the element doesn't have a CSS box, for example because `display: none` is set on the element or one of its parents, then the `currentCSSZoom` is set to 1.

Note that some methods, such as {{domxref("Element.getBoundingClientRect()")}}, return dimensions and position that are relative to the viewport, and hence include the effects of CSS `zoom`.
Other properties and methods return values that are relative to the element itself, and do not include the effects of zooming.
These include, for example, `client*` properties such as {{domxref("Element.clientHeight")}}, `scroll*()` methods like {{domxref("Element.scroll()")}}, and `offset*` properties such as {{domxref("HTMLElement.offsetHeight")}}.
The `currentCSSZoom` property can be used to scale these values to adjust for the effects of zooming.
```

## Value

A number indicating the effective CSS zoom on the element, or 1 if the element is not rendered.

## Examples

This example demonstrates how the `currentCSSZoom` is calculated.

> First we define a nested structure of `<div>` elements where the "parent" is unzoomed and contains a nested element child1 that has `zoom: 2` applied, which in turn contains a nested element "child2" with `zoom: 3` applied.

Reviewer: ```suggestion
First we define a nested structure of `<div>` elements where the "parent" is unzoomed and contains a nested element "child1" that has `zoom: 2` applied, which in turn contains a nested element "child2" with `zoom: 3` applied.
```
